static/js/topic_page/bar_tile.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for rendering a bar tile.
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React, { useEffect, useState } from "react";
+
+import { DataGroup, DataPoint } from "../chart/base";
+import { drawGroupBarChart } from "../chart/draw";
+import { getStatsVarLabel } from "../shared/stats_var_labels";
+import { GetStatSetResponse } from "../tools/shared_util";
+import { StatVarMetadata } from "../types/stat_var";
+import { CHART_HEIGHT } from "./constants";
+import { Point } from "./ranking_tile";
+
+const NUM_PLACES = 6;
+
+const FILTER_STAT_VAR = "Count_Person";
+
+interface BarTilePropType {
+  id: string;
+  title: string;
+  placeDcid: string;
+  enclosedPlaceType: string;
+  statVarMetadata: StatVarMetadata;
+}
+
+export function BarTile(props: BarTilePropType): JSX.Element {
+  const [rawData, setRawData] = useState<GetStatSetResponse | undefined>(null);
+  const [chartData, setChartData] = useState<DataGroup[] | undefined>(null);
+
+  useEffect(() => {
+    fetchData(props, setRawData);
+  }, [props]);
+
+  useEffect(() => {
+    if (rawData) {
+      processData(props, rawData, setChartData);
+    }
+  }, [props, rawData]);
```

Comment:
I think we would only need to re-run processData when rawData changes so I don't think this useEffect needs to also listen for props changes. Same with the following useEffect that does the drawing, I think we only need to re-run draw when the chartData changes

Reply:
This is from the eslint-plugin-react-hooks plugin. There are some discussions about in this thread https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528.

I yet to find sometime to fully understand it, but from initial impression, it's better to include any props, state that this hook refers.

Reply:
oh, interesting, I didn't know that!

Reply:
If a prop change, it should be a re-render from parent component, then this hook should be triggered too.